### PR TITLE
Foundations of a simple but top-level testbench

### PR DESCRIPTION
**Pull request by flaviens**
Here is the foundation of a complete testbench, which for now uses the simplified vision of looking only at write flow, and only the write addresses are looked at (which is functionally incorrect but correctness will come iteratively).

Mostly, I have made enumeration classes robust to alignment issues and managed their packing/unpacking.

### CODE REVIEW

File: simmem_write_only_nocontent/cpp/simmem_axi_structures.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#ifndef SIMMEM_DV_AXI_STRUCTURES
+#define SIMMEM_DV_AXI_STRUCTURES
+
+#include "simmem_axi_dimensions.h"
+
+///////////////////////////
+// Write address request //
+///////////////////////////
+
+class WriteAddressRequest {
```
Comment (GregAC) [inline simmem_write_only_nocontent/cpp/simmem_axi_structures.h]: For these request classes I think you'd be better off with the various fields just being separate public data in the class with a couple of methods that translate from the unpacked to the packed and vice versa. May as well become a struct at this point as it looks more like a plain old data type, so something like this:

```C++
struct WriteAddressRequest {
  uint64_t id;
  uint64_t addr;
  uint64_t burst_length;
  uint64_t burst_size;
  uint64_t burst_type;
  uint64_t lock_type;
  uint64_t memory_type;
  uint64_t protection_type;
  uint64_t qos;

  uint64_t to_packed();
  void from_packed(uint64_t packed_val);
};
```

This avoids lots of repeated code in the get/set methods which makes changing things tricky. If you shuffle the fields around there may be lots to change, especially if you end up needing to pack into something other than a uint64_t.

**Review by GregAC — COMMENTED** [review]
As far as I can tell whilst there is functionality to apply stimulus (i.e. send write requests) and observe response none of this is actually used yet? So this purely provides the outline of the testbench and further work is needed for it to actually stimulate the RTL?

**Comment by flaviens** [comment]
Yes that's right! This PR does not apply or fetch signals. It is meant to get your feedback on the surroundings first!

File: simmem_write_only_nocontent/cpp/simmem_axi_structures.h (hunk 1 above)
Comment (flaviens) [reply simmem_write_only_nocontent/cpp/simmem_axi_structures.h]: Will be done, thanks!

**Comment by flaviens** [comment]
Here is the implementation of it. Improvements will follow as proposed, but this already provides actual delay evaluation and the structure should not move much from its current state.
Integration of read messages will be natural.

**Pull request closed without merging**